syno-plex-update.sh

```diff
@@ -156,12 +160,16 @@ function main {
     write_log "Installed version: ${installed_version}"
     write_log "Latest available version: ${latest_version}"
 
-    if [ ! -z "${latest_version}" ] && [ "${latest_version}" != "${installed_version}" ]; then
+    set +eu
+	/usr/bin/dpkg --compare-versions "$latest_version" gt "$installed_version"
+    if [ "$?" -eq "0" ]; then
+        set -eu
         write_log 'Update available. Trying to download and install'
         notify_update_available
         download_url=$(parse_download_url "${release_meta}")
         download_and_install_package "${download_url}"
     else
+        set -eu
```

Comment:
Using dpkg comparison is a nice improvement 👍  This might look cleaner as a separate function that manages `set (?)eu` internally. Something like:
```bash
function is_latest_version_installed {
    # Check that the installed version is at least as high as the available version.
    local available_version=$1
    local installed_version=$2

    # dpkg version comparison uses exit codes so we'll tolerate errors temporarily.
    set +eu
    /usr/bin/dpkg --compare-versions "$available_version" gt "$installed_version"
    local result="$?"
    set -eu

    echo ${result}
}
```